### PR TITLE
Add pre type validator for setUniform

### PR DESCRIPTION
**Pull request by pandamicro**
Re: https://github.com/cocos/cocos-engine/issues/11774

### Changelog

* Add pre type validator for setUniform

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/render-scene/core/pass.ts [hunk 1]
```diff
@@ -258,6 +258,10 @@ export class Pass {
         const type = Pass.getTypeFromHandle(handle);
         const ofs = Pass.getOffsetFromHandle(handle);
         const block = this._getBlockView(type, binding);
+        if (DEBUG) {
+            const validator = type2validator[type];
+            assertID(validator && validator(value), 12011, binding, Type[type]);
```
Comment (dumganhar) [inline cocos/render-scene/core/pass.ts:263]: After adding validation check, will it break the compatibility of user code? 
If so, we need to add documentation for this change.

File: native/cocos/renderer/core/PassUtils.cpp [hunk 2]
```diff
@@ -225,6 +194,56 @@ const ccstd::unordered_map<gfx::Type, GFXTypeWriterCallback> type2writer = {
...
+         CC_LOG_ERROR("type2validator unknown type");
+         return false;
+     }},
+    {gfx::Type::INT, [](const MaterialProperty &v) -> bool {
+         auto *p = ccstd::get_if<int32_t>(&v);
+         return p != nullptr ? true : false;
+     }},
+    {gfx::Type::INT2, [](const MaterialProperty &v) -> bool {
+         auto *p = ccstd::get_if<Vec2>(&v);
+         return p != nullptr ? true : false;
```
Comment (dumganhar) [inline native/cocos/renderer/core/PassUtils.cpp]: Please fix the errors clang-tidy reported.

File: cocos/render-scene/core/pass.ts (hunk 1 above)
Comment (pandamicro) [reply cocos/render-scene/core/pass.ts:263]: Yes, that's also the reason we must report clear error message for type mismatch.
We can add a note to the release note

**Comment by pandamicro** [comment]
@cocos-robot run test cases

File: native/cocos/renderer/core/PassUtils.cpp (hunk 2 above)
Comment (pandamicro) [reply native/cocos/renderer/core/PassUtils.cpp]: Fixed

**Review by star-e — APPROVED** [review]

**Comment by dumganhar** [comment]
Web test is not passed. Need to be fixed.

**Review by dumganhar — APPROVED** [review]

**Comment by pandamicro** [comment]
> Web test is not passed. Need to be fixed.

Previous test cases has invalid uniform data (Vec4 albedoScale which should be Vec3), this have been fixed by https://github.com/cocos/engine-extensions/pull/189

Need to update editor version for pr test

**Pull request merged**